sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -705,13 +709,16 @@ case class HashAggregateExec(
 
     def outputFromRegularHashMap: String = {
       s"""
-         |while ($iterTerm.next()) {
+         |while ($iterTerm.next()$keepProducingDataCond) {
```

Comment:
Here I only add the stop check for regular hash map. The fast hash map is small and all in memory, it's ok to always output all of it.